src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -271,19 +270,20 @@ internal T GetValue<T>(string key, bool throwEx, out bool found)
 
             if (typeof(T) == typeof(IList<string>))
             {
-                List<string> list = new();
                 if (obj is IList iList)
                 {
-                    foreach (object item in iList)
-                        list.Add(item?.ToString());
+                    string[] arr = new string[iList.Count];
```

Comment:
I didn't want to add null strings.

Reply:
The existing code is adding null strings:
list.Add(item?.ToString())
You're saying the existing code is wrong? I was just maintaining that behavior.

Reply:
@stephentoub let's stay with what we have, i think the existing code could have been improved and will focus on such issues later.
When validating the audience of a token, a null value is not useful.